### ISSUE TITLE
Client login failed when uploading, serious issue

### ISSUE THREAD

**Issue by xml86** [nextcloud/ios#947]
I have 3 Apple devices all running the latest version of iOS 13 (upgraded about 2-3 days ago from iOS 12, the nextcloud iOS client worked fine). This issue started after upgrading to Nextcloud 16.0.5 from 16.0.4 yesterday.

When I upload using the iOS client on any of the devices, the iOS client suddenly stops the upload mid way and starts asking me to login again. The upload also seems to go very slow (even though I'm on the local network as the nextcloud server). 

From the logs, I see multiple messages like this from the 3 devices (I sanitized the some of the data in the log below):

{"reqId":"xxxxxxxxx","level":2,"time":"2019-09-27T14:07:21+00:00","remoteAddr":"192.168.1.xxx","user":"--","app":"core","method":"PROPFIND","url":"\/remote.php\/webdav\/data\/2019\/09","message":"Login failed: 'xxxx' (Remote IP: '192.168.1.xxx')","userAgent":"Mozilla\/5.0 (iOS) Nextcloud-iOS\/2.24.1","version":"16.0.5.1"}

All of the devices have the latest nextcloud iOS client. I even uninstalled and reinstalled to no avail. As soon as I reinstall, login, and start upload, midway it prompts me to login again with the log saying "login failed".

**Comment by small1** [comment]
I seem to have the same problem with one of my clients server.  No clue on why yet tho.

{"reqId":"7ltiIUPQe5YbTVnSilrt","level":2,"time":"2019-10-02T08:47:44+00:00","remoteAddr":"x.x.x.x","user":"--","app":"core","method":"PROPFIND","url":"\/remote.php\/webdav","message":"Login failed: 'xxxxx' (Remote IP: 'x.x.x.x')","userAgent":"Mozilla\/5.0 (iOS) Nextcloud-iOS\/2.24.1","version":"16.0.5.1"}

**Comment by xml86** [comment]
Any updates on this issue? Is no one else other than above 2 users having this issue??

**Comment by poctum** [comment]
I've been having this issue as well. Website and Desktop App (Windows) are working fine.
I did a packet capture on the Nextcloud server during authentication and discovered the authorization: Basic coming across in the HTTP request has a NULL password, which fails authentication and returns 401 to the client. Using an App Token works for a short time and then prompts for username/password again.

user-agent: Mozilla/5.0 (iOS) Nextcloud-iOS/2.24.4

**Comment by brievenbusalex** [comment]
I seem to have the same problem, with the Nextcloud App on iPad 2018 (iOS version 13.1, Nextcloud App 2.25.0, Nextcloud 17.0.1.1):  

Nextcloud logfile:
{"reqId":"R2zggwXxNfqtPJyoeTVO","level":2,"time":"2019-11-25T15:49:19+01:00","remoteAddr":"xx.xxx.x.xx","user":"--","app":"core","method":"PROPFIND","url":"\/remote.php\/webdav","message":"Login failed: 'myname' (Remote IP: 'xx.x.x.xx')","userAgent":"Mozilla\/5.0 (iOS) Nextcloud-iOS\/2.25.0","version":"17.0.1.1"} 

Apache logfile:
xxx.xxx.x.xx - - [25/Nov/2019:15:49:18 +0100] "PROPFIND /remote.php/webdav HTTP/1.1" 401 343
xx.xxx.x.xx - - [25/Nov/2019:15:49:19 +0100] "PROPFIND /remote.php/webdav HTTP/1.1" 401 343
xx.xxx.x.xx - - [25/Nov/2019:15:49:19 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 401 343
xx.xxx.x.xx - - [25/Nov/2019:15:49:20 +0100] "PROPFIND /remote.php/webdav/ HTTP/1.1" 401 343
8

This happens when I dont use the iPad for a while and maybe it had to reset it's session ? 

I have no session settings in my config.php like:

'remember_login_cookie_lifetime'
'session_lifetime'
'session_keepalive'

Can one of theses settings have effect on this problem ?

The app always works again after some retrying, probably gets a new session, but still it's annoying. Anyone idea's / solution ?

**Comment by mentex23** [comment]
I am also facing this issue.. still present on 17.02. any change to get that fixed?

**Comment by whisperdancer** [comment]
Have been having the same issue for a long time. Never gets fixed. Upgraded to 2.25.2 and issue persists. App is unusable on my iPad running IOS 13. I have logged out, removed user, uninstalled, reinstalled. Tried everything. Issue persists. I can use the app for a few minutes at most before it prompts for login. Now it puts a red error dialog at the top and then continually prompts for login. Never experienced this before on any other app in my life. Quite frustrating to say the least. How can this issue not rise to the top of the bug fix list?

This is the error from the logs. It appears multiple times in the log

[webdav] Debug: OCA\DAV\Connector\Sabre\Exception\PasswordLoginForbidden: at <>

    /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Auth/Backend/AbstractBasic.php line 105
    OCA\DAV\Connector\Sabre\Auth->validateUserPass("*** sensitive parameters replaced ***")
    /var/www/nextcloud/apps/dav/lib/Connector/Sabre/Auth.php line 253
    Sabre\DAV\Auth\Backend\AbstractBasic->check(Sabre\HTTP\Reque … "}, Sabre\HTTP\Response {})
    /var/www/nextcloud/apps/dav/lib/Connector/Sabre/Auth.php line 155
    OCA\DAV\Connector\Sabre\Auth->auth(Sabre\HTTP\Reque … "}, Sabre\HTTP\Response {})
    /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Auth/Plugin.php line 200
    OCA\DAV\Connector\Sabre\Auth->check(Sabre\HTTP\Reque … "}, Sabre\HTTP\Response {})
    /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Auth/Plugin.php line 150
    Sabre\DAV\Auth\Plugin->check(Sabre\HTTP\Reque … "}, Sabre\HTTP\Response {})
    <>
    Sabre\DAV\Auth\Plugin->beforeMethod(Sabre\HTTP\Reque … "}, Sabre\HTTP\Response {})
    /var/www/nextcloud/3rdparty/sabre/event/lib/EventEmitterTrait.php line 105
    undefinedundefinedcall_user_func_array([Sabre\DAV\Auth\ … "], [Sabre\HTTP\Requ … }])
    /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 466
    Sabre\Event\EventEmitter->emit(“beforeMethod”, [Sabre\HTTP\Requ … }])
    /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 254
    Sabre\DAV\Server->invokeMethod(Sabre\HTTP\Reque … "}, Sabre\HTTP\Response {})
    /var/www/nextcloud/apps/dav/appinfo/v1/webdav.php line 80
    Sabre\DAV\Server->exec()
    /var/www/nextcloud/remote.php line 163
    undefinedundefinedrequire_once("/var/www/nextcl … p")

PROPFIND /remote.php/webdav/Documents/Logo/pexloe
from XX.XXX.XXX.XXX at 2019-10-06T12:15:36-07:00

**Comment by dschuer** [comment]
Hi,

still having trouble with this bug which cause us problems.
Error with share extension app 2.25.0 #1022 -> pretty happy that this is solved. But there is still a bug in the share extension that is auto configured to the “files” app where you can also access your nextcoud account, using webdav. Usage is sometimes pretty slow, but no errors are shown to the user. But in the background we can see “login failed” that locks out the LDAP User after some time. Which cause even more trouble…

Nextcloud logs are the same as mention here:
Latest App version 2.25.3.6
Nextcloud version: 17.0.0
{"reqId":"VbIk8dduH2WHSSX1IGgn","level":2,"time":"2020-01-15T10:09:02+00:00","remoteAddr":"xxx.xx.xx.xxx","user":"--","app":"core","method":"PROPFIND","url":"\/cloud\/remote.php\/webdav[...]","message":"Login failed: 'username' (Remote IP: 'xxx.xx.xx.xxx')","userAgent":"Mozilla\/5.0 (iOS) Nextcloud-iOS\/2.25.3","version":"17.0.0.9"}

**Comment by hillebr** [comment]
...same error on iOS devices here:
![image](https://user-images.githubusercontent.com/28733806/72601149-29680200-3915-11ea-9f6a-419a533d6bfa.png)

Nextcloud App: 2.25.3.6
Nextcloud Server 17.0.2 (Debian)
User are LDAP authenticated

**Comment by bcutter** [comment]
Someone really just f****d up things - and nothing happens.

Turn to https://github.com/nextcloud/ios/issues/1022#issuecomment-568786092 and ongoing. It´s pure cheek that issue has been CLOSED - having users KNOWING it´s not and giving data to prove that.

I can watch my fail2ban status for nextcloud when trying to upload data to Nextcloud using the Nextcloud app or using the iOS files integration of Nextcloud which registers 5 failed authentication attempts - and blocks that IP.

CAN SOMEBODY with development skills PLEASE FINALLY TAKE THIS SERIOUS ?!?

**Comment by Volker-K** [comment]
Seems to be our problem, too. Accessing NC via NC-App works but using Files App may cause that 401-message.
Users are authenticated via LDAP, passwords have to be changed each 90 days. After entering the new password in NC-App this access works, but using Files App it still seems to be using the old password. It happens even if an application specific password for the NC-App was used.
After deactivating NC in Files App, deleting Files App and installing it from App Store it works again and - as far as I could test - a new password change doesn't cause the error again.

**Comment by GCHPX** [comment]
Same problem here, NC 18 installed, error only happens on iOS device with newest Firmware and app. 
It seems to me this is something with the .htaccess ??

**Comment by bcutter** [comment]
> 
> 
> Same problem here, NC 18 installed, error only happens on iOS device with newest Firmware and app.
> It seems to me this is something with the .htaccess ??

I don´t bet on that (does every NC setup have an .htaccess? thought this is an Apache exclusive thing - I´m running NC on nginx). You see, that´s what we´re all doing here: just guessing. The real experts (developers) are quite silent around here... unfortunately.

**Comment by GCHPX** [comment]
true, but we can just hope that they maybe hear us out...until them it is trial and error....

**Comment by bcutter** [comment]
Yes, but hope is nothing I bet on, seeing this for more than 2 and 1/2 months now. A few days and some app store review adjustments for the app will be made (speaking for myself, just thinking loud).

Okay, back to topic. Here´s another issue, very likely to all of them having the same root cause (authentication process bug)

https://github.com/nextcloud/ios/issues/1113

**Comment by GCHPX** [comment]
soooo, i think i found something that may help the devs..i just deactivated 2fa and removed the account on the app and added it again and the error is gone. can anyone of you guys confirm this? i know this isn´t a solution but my help to narrow it down a bit.

**Comment by dschuer** [comment]
I’m sorry, but we don’t have 2FA in use or installed. But still false Authentications during usage.
In currently the latest version also the described error from #1147. Just opening the app, the error is shown. And two false logins occur (2 LDAP Server configured).
Opening files app to use Nextcloud integration there, also creates false logins…

`{"reqId":"Z9czr0qrk425Iz9Gb6S2","level":2,"time":"2020-02-13T10:43:53+00:00","remoteAddr":"x.x.x.x","user":"--","app":"core","method":"GET","url":"\/cloud\/index.php\/core\/preview.png?file=2020-01-08%2015-42-47-1.PNG&x=281.25&y=500.25&a=1&mode=cover","message":"Login failed: 'my_id' (Remote IP: 'x.x.x.x')","userAgent":"Mozilla\/5.0 (iOS) Nextcloud-iOS\/2.25.7","version":"17.0.2.1"}
{"reqId":"3xVODub4Zzx6acfdek7g","level":2,"time":"2020-02-13T10:43:53+00:00","remoteAddr":"x.x.x.x","user":"--","app":"core","method":"GET","url":"\/cloud\/index.php\/core\/preview.png?file=2020-01-08%2015-42-31-1.PNG&x=281.25&y=500.25&a=1&mode=cover","message":"Login failed: 'my_id' (Remote IP: 'x.x.x.x')","userAgent":"Mozilla\/5.0 (iOS) Nextcloud-iOS\/2.25.7","version":"17.0.2.1"}
{"reqId":"FaNVAO6KY1wgDBJBh5wh","level":2,"time":"2020-02-13T10:43:51+00:00","remoteAddr":"x.x.x.x","user":"--","app":"core","method":"PROPFIND","url":"\/cloud\/remote.php\/webdav\/","message":"Login failed: 'my_id' (Remote IP: 'x.x.x.x')","userAgent":"Mozilla\/5.0 (iOS) Nextcloud-iOS\/2.25.7","version":"17.0.2.1"}
{"reqId":"sUYCFqeg1GiwAJOhoUHm","level":2,"time":"2020-02-13T10:43:49+00:00","remoteAddr":"x.x.x.x","user":"--","app":"core","method":"PROPFIND","url":"\/cloud\/remote.php\/webdav\/","message":"Login failed: 'my_id' (Remote IP: 'x.x.x.x')","userAgent":"Mozilla\/5.0 (iOS) Nextcloud-iOS\/2.25.7","version":"17.0.2.1"}
{"reqId":"YuKwDp6do0ip8SihNKTE","level":2,"time":"2020-02-13T10:43:47+00:00","remoteAddr":"x.x.x.x","user":"--","app":"core","method":"PROPFIND","url":"\/cloud\/remote.php\/webdav","message":"Login failed: 'my_id' (Remote IP: 'x.x.x.x')","userAgent":"Mozilla\/5.0 (iOS) Nextcloud-iOS\/2.25.7","version":"17.0.2.1"}
{"reqId":"lJ3SdzYsPndefi3FwVLP","level":2,"time":"2020-02-13T10:43:46+00:00","remoteAddr":"x.x.x.x","user":"--","app":"core","method":"PROPFIND","url":"\/cloud\/remote.php\/webdav","message":"Login failed: 'my_id' (Remote IP: 'x.x.x.x')","userAgent":"Mozilla\/5.0 (iOS) Nextcloud-iOS\/2.25.7","version":"17.0.2.1"}
{"reqId":"KN51xdRaToHERgDt0CsW","level":2,"time":"2020-02-13T10:43:42+00:00","remoteAddr":"x.x.x.x","user":"--","app":"core","method":"PROPFIND","url":"\/cloud\/remote.php\/webdav","message":"Login failed: 'my_id' (Remote IP: 'x.x.x.x')","userAgent":"Mozilla\/5.0 (iOS) Nextcloud-iOS\/2.25.7","version":"17.0.2.1"}
{"reqId":"1yNRrP4DPGxxfrd97ptw","level":2,"time":"2020-02-13T10:43:40+00:00","remoteAddr":"x.x.x.x","user":"--","app":"core","method":"PROPFIND","url":"\/cloud\/remote.php\/webdav","message":"Login failed: 'my_id' (Remote IP: 'x.x.x.x')","userAgent":"Mozilla\/5.0 (iOS) Nextcloud-iOS\/2.25.7","version":"17.0.2.1"}
{"reqId":"zRgNMk03dSu757rbC4Ga","level":2,"time":"2020-02-13T10:39:17+00:00","remoteAddr":"x.x.x.x","user":"--","app":"core","method":"PROPFIND","url":"\/cloud\/remote.php\/webdav","message":"Login failed: 'my_id' (Remote IP: 'x.x.x.x')","userAgent":"Mozilla\/5.0 (iOS) Nextcloud-iOS\/2.25.7","version":"17.0.2.1"}
{"reqId":"Y1VXK6bwExtQ20bJGlUi","level":2,"time":"2020-02-13T10:39:15+00:00","remoteAddr":"x.x.x.x","user":"--","app":"core","method":"PROPFIND","url":"\/cloud\/remote.php\/webdav","message":"Login failed: 'my_id' (Remote IP: 'x.x.x.x')","userAgent":"Mozilla\/5.0 (iOS) Nextcloud-iOS\/2.25.7","version":"17.0.2.1"}
{"reqId":"EbhNRc15leRxqkrqQef6","level":2,"time":"2020-02-13T10:37:48+00:00","remoteAddr":"x.x.x.x","user":"--","app":"core","method":"PROPFIND","url":"\/cloud\/remote.php\/webdav","message":"Login failed: 'my_id' (Remote IP: 'x.x.x.x')","userAgent":"Mozilla\/5.0 (iOS) Nextcloud-iOS\/2.25.7","version":"17.0.2.1"}
{"reqId":"XIS06Jn6jljnuGY7t5hj","level":2,"time":"2020-02-13T10:37:46+00:00","remoteAddr":"x.x.x.x","user":"--","app":"core","method":"PROPFIND","url":"\/cloud\/remote.php\/webdav","message":"Login failed: 'my_id' (Remote IP: 'x.x.x.x')","userAgent":"Mozilla\/5.0 (iOS) Nextcloud-iOS\/2.25.7","version":"17.0.2.1"}
`

**Comment by dschuer** [comment]
Release notes of Version 2.25.8
- Fixed the use of Nextcloud with LDAP
I’m glad to confirm that **for us version 2.25.8 fixed this bug** and a lot of less failed logins are logged. With latest version only a few, no longer a bunch of failed logins from one user caused by usage of Nextcloud. Thank you!
